src/jit/inlinepolicy.cpp

```diff
@@ -103,6 +103,39 @@ void LegacyPolicy::noteBool(InlineObservation obs, bool value)
             // Passed the profitability screen. Update candidacy.
             setCandidate(obs);
             break;
+        case InlineObservation::CALLEE_BEGIN_OPCODE_SCAN:
+            {
+                // Set up the state machine, if this inline is
+                // discretionary and is still a candidate.
+                if (inlDecisionIsCandidate(inlDecision)
+                    && (inlObservation == InlineObservation::CALLEE_IS_DISCRETIONARY_INLINE))
+                {
+                    // Better not have a state machine already.
+                    assert(inlStateMachine == nullptr);
+                    inlStateMachine = new (inlCompiler, CMK_Inlining) CodeSeqSM;
```

Comment:
It appears previously we just get the address from local or object's instance field, but now you create one from the heap. Do we have concern about repeating this step?